system/jlib/jstats.cpp

```diff
@@ -396,6 +400,43 @@ static StringBuffer & formatIPV4(StringBuffer & out, unsigned __int64 value)
     return out.appendf("%d.%d.%d.%d", ip1, ip2, ip3, ip4);
 }
 
+static const char *getMoneyLocale()
+{
+    static bool haveLocale = false;
+    static StringAttr moneyLocale;
+
+    if (!haveLocale)
+    {
+        StringBuffer localestr;
+        if (isGlobalConfigReady())
+        {
+            queryGlobalConfig().getProp("cost/@moneyLocale", localestr);
+            haveLocale = true;
+            try // try setting locale, so exceptions can be caught here
+            {
+                std::locale(localestr.str());
+            }
+            catch(...)
+            {
+                OERRLOG("moneyLocale invalid: %s", localestr.str());
+                localestr.clear();
+            }
+        }
+        // note that an empty string causes default locale to used
+        moneyLocale.set(localestr.str());
+    }
+
+    return moneyLocale.get();
+}
+
+static StringBuffer & formatMoney(StringBuffer &out, unsigned __int64 value)
+{
+    std::stringstream ss;
+    ss.imbue(std::locale(getMoneyLocale()));
```

Comment:
better to have a static std::locale which is the value set up by getMoneyLocale to avoid the cost of recreating the locale object.